lib/ansible/modules/cloud/amazon/aws_waf_condition.py

```diff
@@ -580,7 +583,13 @@ def find_and_update_condition(self, condition_set_id):
             update['Updates'] = missing + extra
             func = getattr(self.client, 'update_' + self.method_suffix)
             try:
-                run_func_with_change_token_backoff(self.client, self.module, update, func)
+                result = run_func_with_change_token_backoff(self.client, self.module, update, func, wait=True)
+            except botocore.exceptions.WaiterError as e:
+                self.module.fail_json_aws(
+                    e,
+                    msg='Change token did not sync, or took too long %s : %s' % (
```

Comment:
This line is problematic because `result` does not get set if the exception occurs.

Reply:
Oh, good catch. I think that's why I took it out in the other two. I'll patch this.